torch/nn/parameter.py

```diff
@@ -42,3 +42,9 @@ def __reduce_ex__(self, proto):
             torch._utils._rebuild_parameter,
             (self.data, self.requires_grad, OrderedDict())
         )
+
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+        with DisableTorchFunction():
+            return func(*args, **kwargs)
```

Comment:
This seems like it would merit an explanation about why the override is necessary (operations on Parameter never return Parameter). Additionally, it seems suboptimal that all operations on Parameter now hit a Python-implemented `__torch_function__`, where previously they didn't. This case probably needs to be explicitly optimized for in the C++.

Reply:
I.e., we should have some fast path for subclasses of Tensor that don't return subclass by default when operations are run, AKA the legacy behavior. We need a way for people who previously subclassed Tensor to opt into this behavior!

Reply:
(Or perhaps opt out? As the default subclass behavior is BC-breaking)

Reply:
This is my oversight, I should have realized this would break Parameter in the RFC:

> torch.Tensor will gain a generic __torch_function__ of the following form:

If we're going to change this behavior by default, we need to assess the impact on other libraries which subclass Tensor today. I am pretty sure there are subclassed tensors in the wild. So we need to keep the old behavior, have a flag for the new behavior, and we can do a deprecation cycle if we want to change the beahvior.

Reply:
If you allow me to push back a little: The default `__torch_function__` does more-or-less exactly what we want: Preserve subclasses of `Tensor` on method/function calls. Without it, the whole RFC will be delayed by the deprecation period.

What I've done instead is provide an opt-out, if one needs it, by setting `__torch_function__` to `None`. I've done this in `Parameter` as well, so it won't bounce back to Python. (even though using it on parameter had the somewhat-intended side effect of testing `__torch_function__` thoroughly.)